crt_portal/cts_forms/views.py

```diff
@@ -936,61 +935,160 @@ def get(self, request, id):
 
 class DispositionActionsView(LoginRequiredMixin, FormView):
     """ CRT view to update report disposition"""
+    EMPTY_CHOICE = 'Multiple'
 
-    def get(self, request):
+    def get_shared_report_values(self, record_query, keys):
+        """
+        Given a record query and a list of keys, determine if a key has a
+        singular value within that query. Used to set initial fields
+        for bulk update forms.
+        """
+        # make sure the queryset does not order by anything, otherwise
+        # we will have difficulty getting distinct results.
+        query = record_query.order_by()
+        for key in keys:
+            values = query.values_list(key, flat=True).distinct()
+            if values.count() == 1:
+                if key == 'retention_schedule':
+                    yield key, RetentionSchedule.objects.get(retention_years=values[0]).name
+                else:
+                    yield key, values[0]
+            else:
+                yield key, self.EMPTY_CHOICE
```

Comment:
`continue` might be the key to unnesting this

```suggestion
        for key in keys:
            values = query.values_list(key, flat=True).distinct()
            if values.count() != 1:
                yield key, self.EMPTY_CHOICE
                continue
            if key == 'retention_schedule':
                yield key, RetentionSchedule.objects.get(retention_years=values[0]).name
                continue
            yield key, values[0]
```